frontend/src/app/search-city.service.ts

```diff
@@ -22,4 +22,11 @@ export class SearchCityService {
     return this.http.get('/api/cities', { params : new HttpParams().set('query', term)});
   }
 
+  uploadCities(data: string): Observable<HttpEvent<any>> {
+    const req = new HttpRequest('POST', '/api/cities/uploads', data, {
+      reportProgress: true,
+    });
+
+    return this.http.request(req);
```

Comment:
Nice! Is it mandatory to do a `request` here? not possible with a simple `post`?